src/lib/storage/fake_device_factory.rb

```diff
@@ -212,9 +212,9 @@ def set_topology_attributes!(disk, args)
         io_size = args["io_size"]
         min_grain = args["min_grain"]
         align_ofs = args["align_ofs"]
-        disk.topology.optimal_io_size = io_size.size if io_size && io_size.size > 0
+        disk.topology.optimal_io_size = io_size.size if io_size && !io_size.empty?
```

Comment:
I agree, that size looks strange here, but also as @lslezak mention, `io_size.size` It looks like io_size is not good name...maybe it is number of available blocks? what it actually strands for?